akka-kafka-to-sharding/src/main/resources/application.conf

```diff
@@ -0,0 +1,35 @@
+akka.actor {
+  provider = "cluster"
+  // TODO remove once protobuf message for ShardLocation
+  allow-java-serialization = on
+}
+
+akka {
+
+  remote.artery {
+    canonical {
+      hostname = "127.0.0.1"
+    }
+  }
+
+  cluster {
+
+    seed-nodes = [
+      "akka://KafkaToSharding@127.0.0.1:2551",
+      "akka://KafkaToSharding@127.0.0.1:2552"]
+
+    sharding {
+      # set to the same number of kafka partitions.
+      # alternatively there needs to be a mapping from kafka partition to
+      # shard
+      number-of-shards = 128
```

Comment:
I think I'm missing something here.  If we have an unequal number of partitions to shards why do we need to define a map?  I'm probably oversimplifying this, but could we derive a map by taking locally assigned partitions and mapping them to locally available shards automatically?  When the cluster first forms and a Kafka consumer group rebalance is complete we have all Kafka partitions assigned to available nodes.  When a message is consumed it passes through the entity id extractor that will only route to a locally available shard.  At the same time we could update a ddata map that to assign the messages partition to the shard it was sent to.

Some other things.  Sometimes the number of partitions a consumer group has is not obvious.  For example, you could consume two topics at the same time, or an arbitrary set of partitions across as many topics as you like.

Or maybe you want to setup more than 1 consumer per node (to consume a different topic, for some different workload, but you want to process it on the same cluster).  I'm not sure how you would define the number of shards in advance for a situation like that.

Reply:
> but could we derive a map by taking locally assigned partitions and mapping them to locally available shards automatically

I could be missing something we could do automagically but I don't think that's possible. An entity always has to belong to the same shard, we can't update it dynamically. 

There are no locally available shards initially, they are only created when messages pass through and that mapping from entity id to shard is a pure function, it can't for instance query ddata.

So for all of this to glue together the mapping between partitions and shards has to be fixed and all messages for the same entity have to come in the same partition.

Reply:
> Some other things. Sometimes the number of partitions a consumer group has is not obvious. For example, you could consume two topics at the same time, or an arbitrary set of partitions across as many topics as you like.

I haven't thought about scenarios like this yet, but as we're not providing the glue, it is only a sample showing how to use dynamic sharding. I think users can set this up.